### PR TITLE
refactor: Support batch export models as views 

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

This PR sets the ground work to support person batch exports by introducing a set of parametrized views that will serve as the basis of batch exports moving forward. This new set of views includes a view for the persons model. Using parametrized views over the old approach has the benefit of being easier to parse the query logic when it's laid out in plain SQL (ideally this should be SQL files, but our migration system works with Python files...).

## Changes

* Introduce new parametrized views for person and events batch exports.
  * The person model is completely new, so no impact on existing functionality. This PR only introduces the view of the model, a follow-up PR will make use of it.
  * On the other hand, the event model is changing slightly with the new view:
     * We are now using `GROUP BY` instead of `DISTINCT`. This should perform better and use less memory (in particular with `optimize_aggregation_in_order` enabled).
     * We are now using a `PREWHERE` clause for filtering first on `inserted_at`. Once again, this should result in better performance.
     * Event model is divided into 3: "default", "unbounded", and "backfill", with one view for each. Determined by how they use `timestamp` and `inserted_at` filters. Should behave the same as before, but laid out in plain SQL instead of composed via Python formatting.
     * For legacy reasons, event model is different according to destination. Eventually, I'll create views for each destination to also lay them out in SQL
  * Using the views required refactoring large portions of code.

* Introduce new methods to iterate over records asynchronously.
  * This requires implementing parsing of `pyarrow.RecordBatch`.
  * Will initially only use these to iterate over records of new models (i.e. persons), and switch to events on a later date.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

* Updated a few tests to work with views.
* Ran all unit tests.

### CODE REVIEW

File: posthog/temporal/tests/batch_exports/conftest.py [hunk 1]
```diff
@@ -123,3 +138,20 @@ async def setup_postgres_test_db(postgres_config):
         await cursor.execute(sql.SQL("DROP DATABASE {}").format(sql.Identifier(postgres_config["database"])))
 
     await connection.close()
+
+
+@pytest_asyncio.fixture(scope="module", autouse=True)
+async def create_batch_export_views(clickhouse_client, django_db_setup):
```
Comment (tomasfarias) [inline posthog/temporal/tests/batch_exports/conftest.py:144]: Apparently clickhouse migrations don't really run in tests.

File: posthog/api/test/test_app_metrics.py [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import datetime as dt
 import json
-import uuid
```
Comment (tomasfarias) [inline posthog/api/test/test_app_metrics.py:3]: Every change on this file is just cleaning up.

File: posthog/temporal/common/clickhouse.py [hunk 3]
```diff
@@ -357,6 +359,23 @@ def stream_query_as_arrow(
             with pa.ipc.open_stream(pa.PythonFile(response.raw)) as reader:
                 yield from reader
 
+    async def astream_query_as_arrow(
```
Comment (tomasfarias) [inline posthog/temporal/common/clickhouse.py:362]: Not in use yet, will use for person model exports.

File: posthog/temporal/batch_exports/batch_exports.py [hunk 4]
```diff
@@ -193,56 +262,69 @@ def iter_records(
...
+) -> AsyncRecordsGenerator:
+    if model == "persons":
+        view = SELECT_FROM_PERSONS_VIEW
+    else:
+        # TODO: Let this model be exported by `astream_query_as_arrow`.
+        # Just to reduce risk, I don't want to change the function that runs 100% of the exports
+        # without battle testing it first.
+        # There are already changes going out to the queries themselves that will impact events in a
+        # positive way. So, we can come back later and drop this block.
+        for record_batch in iter_records(client, team_id=team_id, is_backfill=is_backfill, **parameters):
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/batch_exports.py:322]: See TODO above: For now, keeping old functionality intact for safety. Will release person model first, test, and then switch events.

File: posthog/temporal/batch_exports/batch_exports.py [hunk 5]
```diff
@@ -77,78 +77,20 @@ def get_timestamp_field(is_backfill: bool) -> str:
     return timestamp_field
 
 
-async def get_rows_count(
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/batch_exports.py:80]: We don't count rows anymore, so this function is unused. Cleaning up.

File: posthog/temporal/batch_exports/batch_exports.py [hunk 6]
```diff
@@ -30,136 +30,120 @@
...
+FROM
+    events_batch_export(
+        team_id={team_id},
+        lookback_days={lookback_days},
+        interval_start={interval_start},
+        interval_end={interval_end},
+        include_events={include_events}::Array(String),
+        exclude_events={exclude_events}::Array(String)
     )
+FORMAT ArrowStream
```
Comment (fuziontech) [inline posthog/temporal/batch_exports/batch_exports.py:62]: This is exciting. 
https://clickhouse.com/docs/en/integrations/data-formats/arrow-avro-orc#arrow-data-streaming

File: posthog/temporal/batch_exports/batch_exports.py [hunk 7]
```diff
@@ -30,136 +30,120 @@
...
+        # TODO: Let this model be exported by `astream_query_as_arrow`.
+        # Just to reduce risk, I don't want to change the function that runs 100% of the exports
+        # without battle testing it first.
+        # There are already changes going out to the queries themselves that will impact events in a
+        # positive way. So, we can come back later and drop this block.
+        for record_batch in iter_records(client, team_id=team_id, is_backfill=is_backfill, **parameters):
+            yield record_batch
+        return
+
+    async for record_batch in client.astream_query_as_arrow(view, query_parameters=parameters):
```
Comment (fuziontech) [inline posthog/temporal/batch_exports/batch_exports.py:145]: nice.

**Review by fuziontech — APPROVED** [review]
This is great. I'm excited to use Views here...easy to iterate and low risk.

**Pull request merged**